app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/ImportationFilesManager.kt

```diff
@@ -0,0 +1,280 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.screen.newtransfer
+
+import android.content.ContentResolver
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.provider.OpenableColumns
+import android.util.Log
+import androidx.core.net.toUri
+import com.infomaniak.swisstransfer.ui.components.FileUi
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.NewTransferViewModel.Companion.LOCAL_COPY_FOLDER
+import com.infomaniak.swisstransfer.ui.utils.FileNameUtils
+import dagger.hilt.android.qualifiers.ApplicationContext
+import io.sentry.Sentry
+import io.sentry.SentryLevel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ImportationFilesManager @Inject constructor(@ApplicationContext private val appContext: Context) {
+
+    private val filesToImport: TransferCountChannel = TransferCountChannel()
+    val filesToImportCount: StateFlow<Int> = filesToImport.count
+    val currentSessionTotalUploadedFiles: StateFlow<Int> = filesToImport.currentSessionTotalUploadedFiles
+
+    private val _importedFiles = FilesMutableStateFlow()
+    val importedFiles = _importedFiles.flow
+
+    private val _failedFiles = MutableSharedFlow<PickedFile>()
+    val failedFiles = _failedFiles.asSharedFlow()
+
+    // Importing a file locally can take up time. We can't base the list of already used names on _importedFiles's value because a
+    // new import with the same name could occur while the file is still importing. This would lead to a name collision.
+    // This list needs to mark a name as "taken" as soon as the file is queued to be imported and until the file is removed from
+    // the list of already imported files we listen to in the LazyRow.
```

Comment:
```suggestion
    /**
     * Importing a file locally can take up time. We can't base the list of already used names on _importedFiles's value because
     * a new import with the same name could occur while the file is still importing. This would lead to a name collision.
     * This list needs to mark a name as "taken" as soon as the file is queued to be imported and until the file is removed
     * from the list of already imported files we listen to in the LazyRow.
     */
```

Reply:
I think this would be more fitting as a "gray" comment rather than a kDoc